File: src/composables/deduty/package/scheme/package.ts
import type { IDedutyFileCollection } from './files'

export interface IDedutyPackageMeta {
  name: string
  version: string
  language: string
  tags: string[]
}

export interface IDedutyPackage {
  id: string
  meta: IDedutyPackageMeta
  files: IDedutyFileCollection
}

File: src-tauri/crates/deduty_package_traits/src/package.rs
use std::any::Any;

use crate::file::DedutyFileCollection;
use crate::lection::DedutyLection;


pub trait DedutyPackageMeta: Sync + Send {
    fn name(&self) -> String;
    fn version(&self) -> String;
    fn language(&self) -> String;
    fn tags(&self) -> Vec<String>;
}

pub trait DedutyPackage: Any + Sync + Send {
    fn as_any_ref(&self) -> &(dyn Any + Send + Sync);
    fn as_any_mut(&mut self) -> &mut (dyn Any + Send + Sync);

    fn id(&self) -> String;
    fn meta(&self) -> &dyn DedutyPackageMeta;
    fn files(&self) -> &dyn DedutyFileCollection;
    fn lections(&self) -> Vec<&dyn DedutyLection>;
}
